ssl/quic/quic_impl.c

```diff
@@ -1318,34 +1323,36 @@ QUIC_TAKES_LOCK
 int ossl_quic_get_event_timeout(SSL *s, struct timeval *tv, int *is_infinite)
 {
     QCTX ctx;
-    OSSL_TIME deadline = ossl_time_infinite();
+    QUIC_REACTOR *reactor;
+    OSSL_TIME deadline;
+    OSSL_TIME basetime;
 
-    if (!expect_quic_cs(s, &ctx))
+    if (!expect_quic_any(s, &ctx))
         return 0;
 
     qctx_lock(&ctx);
 
-    if (ctx.qc->started)
-        deadline
-            = ossl_quic_reactor_get_tick_deadline(ossl_quic_channel_get_reactor(ctx.qc->ch));
-
-    if (ossl_time_is_infinite(deadline)) {
-        *is_infinite = 1;
-
-        /*
-         * Robustness against faulty applications that don't check *is_infinite;
-         * harmless long timeout.
-         */
-        tv->tv_sec  = 1000000;
-        tv->tv_usec = 0;
-
-        qctx_unlock(&ctx);
-        return 1;
+    reactor = ossl_quic_obj_get0_reactor(ctx.obj);
+    /*
+     * TODO: There's no longer a way to indicate infinite timeout, should this
```

Comment:
Not sure why you removed the previous code for checking if the deadline was infinite (but yes arguably it could be moved to `ossl_quic_reactor_get_tick_deadline`....but probably not as part of this rebase)

Reply:
Restored.